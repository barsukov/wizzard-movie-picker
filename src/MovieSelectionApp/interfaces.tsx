export interface IState {
  episodes: Array<any>;
  favorites: Array<any>;
}
export interface IHomePageProps{
  episodes: Array<IEpisode>,
  toogleFav: (dispatch: any, episode: IEpisode, favorites: IEpisode[]) => IAction,
  favorites: Array<IEpisode>
}
export type Dispatch = React.Dispatch<IAction>;

export interface IEpisode {
  id: string;
  name: string;
  image: any;
}

export interface IAction {
  type: string;
  payload: any;
}
