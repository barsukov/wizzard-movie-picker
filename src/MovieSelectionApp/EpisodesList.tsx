import React from "react";
import { IEpisode, Dispatch, IState, FavAction } from "./interfaces";
import { Button, Image, Heading, Card, Text } from "rebass";
interface IProps {
  episodes: Array<IEpisode>;
  toggleFavAction: FavAction;
  favourites: Array<IEpisode>;
  store: { state: IState; dispatch: Dispatch };
}

export default function EpisodesList(props: IProps): Array<JSX.Element> {
  const { episodes, toggleFavAction, favourites, store } = props;
  const { state, dispatch } = store;

  return episodes.map((episode: IEpisode) => {
    return (
      <Card>
        <Image
          sx={{
            width: ["100%", "100%"],
            borderRadius: 8
          }}
          src={!!episode.image ? episode.image.medium : ""}
          alt={`Rick and Mort ${episode.name}`}
        />
        <Heading>
          {" "}
          Seasion: {episode.season} Number: {episode.number}
        </Heading>
        <Text>{episode.name}</Text>
        <Button
          variant="primary"
          onClick={() => toggleFavAction(state, dispatch, episode)}
        >
          {favourites.find((fav: IEpisode) => fav.id === episode.id)
            ? "Unfav"
            : "Fav"}
        </Button>
      </Card>
    );
  });
}
