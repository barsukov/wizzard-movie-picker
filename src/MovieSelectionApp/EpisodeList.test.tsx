import * as React from "react";
import { shallow, mount } from "enzyme";
import EpisodesList from "./EpisodesList";
import { IEpisodeListProps, IState, IEpisode } from "./interfaces";
test("EpisodesList renders without elements", () => {
  const TestEpisode: IEpisode = {
    airdate: "test",
    airstamp: "test",
    airtime: "test",
    id: 1,
    image: { medium: "test", original: "originalTest" },
    name: "Bob",
    number: 2,
    runtime: 3,
    season: 4,
    summary: "Good Episode",
    url: "GoodUrl"
  };
  const initialState: IState = {
    episodes: [TestEpisode],
    favourites: []
  };

  const store = { state: initialState, dispatch: jest.fn() };
  const props: IEpisodeListProps = {
    ...store.state,
    toggleFavAction: jest.fn(),
    store: store
  };

  const List: any = mount(<EpisodesList {...props} />);

  expect(List).toMatchSnapshot();
});
