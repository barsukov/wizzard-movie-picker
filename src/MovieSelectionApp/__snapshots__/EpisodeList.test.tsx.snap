// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesList renders without elements 1`] = `
<EpisodesList
  episodes={
    Array [
      Object {
        "airdate": "test",
        "airstamp": "test",
        "airtime": "test",
        "id": 1,
        "image": Object {
          "medium": "test",
          "original": "originalTest",
        },
        "name": "Bob",
        "number": 2,
        "runtime": 3,
        "season": 4,
        "summary": "Good Episode",
        "url": "GoodUrl",
      },
    ]
  }
  favourites={Array []}
  store={
    Object {
      "dispatch": [MockFunction],
      "state": Object {
        "episodes": Array [
          Object {
            "airdate": "test",
            "airstamp": "test",
            "airtime": "test",
            "id": 1,
            "image": Object {
              "medium": "test",
              "original": "originalTest",
            },
            "name": "Bob",
            "number": 2,
            "runtime": 3,
            "season": 4,
            "summary": "Good Episode",
            "url": "GoodUrl",
          },
        ],
        "favourites": Array [],
      },
    }
  }
  toggleFavAction={[MockFunction]}
>
  <ForwardRef
    key="1"
  >
    <Styled(div)
      variant="card"
    >
      <div
        className="css-vurnku"
      >
        <Styled(div)
          sx={
            Object {
              "backgroundImage": "url(originalTest)",
              "backgroundSize": "cover",
              "bg": "gray",
              "borderRadius": 8,
              "color": "white",
              "px": 4,
              "py": 6,
            }
          }
        >
          <div
            className="css-h6swki"
          >
            <Styled(div)
              sx={
                Object {
                  "backgroundSize": "cover",
                  "bg": "white",
                  "borderRadius": 8,
                  "color": "white",
                  "opacity": "60%",
                  "px": 1,
                  "py": 3,
                }
              }
            >
              <div
                className="css-1iybr07"
              >
                <ForwardRef
                  color="black"
                  fontSize={
                    Array [
                      3,
                      5,
                    ]
                  }
                  textAlign="center"
                >
                  <Styled(div)
                    __css={
                      Object {
                        "fontFamily": "heading",
                        "fontSize": 4,
                        "fontWeight": "heading",
                        "lineHeight": "heading",
                      }
                    }
                    as="h2"
                    color="black"
                    fontSize={
                      Array [
                        3,
                        5,
                      ]
                    }
                    textAlign="center"
                    tx="text"
                    variant="heading"
                  >
                    <h2
                      className="css-1agp5a6"
                    >
                      Seasion: 
                      4
                       Number: 
                      2
                    </h2>
                  </Styled(div)>
                </ForwardRef>
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
        <Styled(div)
          episodes={
            Array [
              Object {
                "airdate": "test",
                "airstamp": "test",
                "airtime": "test",
                "id": 1,
                "image": Object {
                  "medium": "test",
                  "original": "originalTest",
                },
                "name": "Bob",
                "number": 2,
                "runtime": 3,
                "season": 4,
                "summary": "Good Episode",
                "url": "GoodUrl",
              },
            ]
          }
          favourites={Array []}
          store={
            Object {
              "dispatch": [MockFunction],
              "state": Object {
                "episodes": Array [
                  Object {
                    "airdate": "test",
                    "airstamp": "test",
                    "airtime": "test",
                    "id": 1,
                    "image": Object {
                      "medium": "test",
                      "original": "originalTest",
                    },
                    "name": "Bob",
                    "number": 2,
                    "runtime": 3,
                    "season": 4,
                    "summary": "Good Episode",
                    "url": "GoodUrl",
                  },
                ],
                "favourites": Array [],
              },
            }
          }
          sx={
            Object {
              "display": "grid",
              "gridGap": 2,
            }
          }
          toggleFavAction={[MockFunction]}
        >
          <div
            className="css-vl1zvk"
          >
            <ForwardRef
              color="primary"
              fontSize={
                Array [
                  3,
                  4,
                  5,
                ]
              }
              fontWeight="bold"
            >
              <Styled(div)
                color="primary"
                fontSize={
                  Array [
                    3,
                    4,
                    5,
                  ]
                }
                fontWeight="bold"
                tx="text"
              >
                <div
                  className="css-911jbo"
                >
                  Bob
                </div>
              </Styled(div)>
            </ForwardRef>
            <ForwardRef
              onClick={[Function]}
              variant="secondary"
            >
              <Styled(div)
                __css={
                  Object {
                    "appearance": "none",
                    "bg": "primary",
                    "border": 0,
                    "borderRadius": 4,
                    "color": "white",
                    "display": "inline-block",
                    "fontSize": "inherit",
                    "lineHeight": "inherit",
                    "px": 3,
                    "py": 2,
                    "textAlign": "center",
                    "textDecoration": "none",
                  }
                }
                as="button"
                onClick={[Function]}
                tx="buttons"
                variant="secondary"
              >
                <button
                  className="css-vhv7r1"
                  onClick={[Function]}
                >
                  Fav
                </button>
              </Styled(div)>
            </ForwardRef>
          </div>
        </Styled(div)>
      </div>
    </Styled(div)>
  </ForwardRef>
</EpisodesList>
`;
